R/jointRPCA.R

```diff
@@ -0,0 +1,616 @@
+    tables <- list(assay(data))
+    nm <- tryCatch(assayNames(data)[1], error = function(e) NULL)
+    names(tables) <- if (is.null(nm) || is.na(nm)) "assay1" else nm  
+  } else if (is.list(data) && all(sapply(data, is.matrix))) {
+    tables <- data
+    if (is.null(names(tables))) names(tables) <- paste0("view", seq_along(tables))  
+  } else if (is.matrix(data)) {
+    tables <- list(data); names(tables) <- "assay1"  
+  } else {
+    stop("Unsupported input type ...")
+  }
+  
+  jointRPCA(tables = tables, ...)
+}
+
+#' Joint Robust PCA on Multiple Compositional Tables
+#'
+#' Performs Joint Robust Principal Component Analysis (RPCA) using OptSpace on multiple compositional tables.
+#' Automatically handles shared sample alignment, optional train/test splitting, and compositional preprocessing.
+#'
+#' @param tables A list of compositional data tables (matrices or data frames).
+#'        For automatic handling of Bioconductor objects (e.g., MultiAssayExperiment, 
+#'        TreeSummarizedExperiment, SummarizedExperiment), or single matrices, 
+#'        consider using `jointRPCAuniversal()` instead.
+#' @param n.test.samples Integer specifying the number of samples to hold out for testing (only used if `sample.metadata` is NULL). Default is 10.
+#' @param sample.metadata Optional data frame containing sample-level metadata. Must include a column indicating train/test labels.
+#' @param train.test.column The name of the column in `sample.metadata` that defines training vs test samples.
+#' @param n.components Integer specifying the number of principal components to compute. Must be at least 2.
+#' @param rclr.transform.tables Logical; whether to apply rclr transformation to each input table before ordination. Default is TRUE.
+#' @param min.sample.count Minimum total count required for a sample to be retained during filtering. Default is 0 (no filtering).
+#' @param min.feature.count Minimum total count required for a feature to be retained. Default is 0 (no filtering).
+#' @param min.feature.frequency Minimum percentage (0–100) of samples in which a feature must be non-zero to be retained. Default is 0.
+#' @param max.iterations Maximum number of optimization iterations. Must be at least 1.
+#'
+#' @return A list with:
+#' \describe{
+#'   \item{ord.res}{An \code{OrdinationResults} object containing sample scores, feature loadings, and variance explained.}
+#'   \item{U.dist.res}{A distance matrix (samples × samples) based on the learned sample embeddings.}
+#'   \item{cv.dist}{A data frame of cross-validation error statistics across iterations.}
+#'   \item{rclr.tables}{A list of CLR-transformed tables used as input for the ordination. These are useful for accessing dataset-specific embeddings or for reuse in downstream analyses.}
+#' }
+#'
+#' @examples
+#' \dontrun{
+#' # See examples/joint_rpca_example.qmd for a full reproducible test case.
+#' }
+#'
+#' @export
+
+jointRPCA <- function(tables,
```

Comment:
Do we need to export jointRPCA, or can we already do everything with jointRPCAuniversal function? I would avoid exporting two rather similar methods.

I suggest to check how mia functions in R/getCrossAssociation.R were implemented. JointRPCA case is different in that it can have more than 2 experiments, so your solution to provide a list of experiments is good.

Note that each experiment in MAE can have multiple assays. Your pipeline currently picks the default assay from each experiment. It would be good to provide a list of experiments, and the list of assays. Probably good to require that the user does this explicitly for every case, then it will be more unambiguous(?)